shell/platform/windows/accessibility_root_node.h

```diff
@@ -0,0 +1,121 @@
+                             LONG* width,
+                             LONG* height,
+                             VARIANT var_id) override;
+
+  // Traverses to another UI element and retrieves the object.
+  IFACEMETHODIMP accNavigate(LONG nav_dir,
+                             VARIANT start,
+                             VARIANT* end) override;
+
+  // Retrieves an IDispatch interface pointer for the specified child.
+  IFACEMETHODIMP get_accChild(VARIANT var_child,
+                              IDispatch** disp_child) override;
+
+  // Retrieves the number of accessible children.
+  IFACEMETHODIMP get_accChildCount(LONG* child_count) override;
+
+  // Retrieves a string that describes the object's default action.
+  IFACEMETHODIMP get_accDefaultAction(VARIANT var_id,
+                                      BSTR* default_action) override;
+
+  // Retrieves the tooltip description.
+  IFACEMETHODIMP get_accDescription(VARIANT var_id, BSTR* desc) override;
+
+  // Retrieves the object that has the keyboard focus.
+  IFACEMETHODIMP get_accFocus(VARIANT* focus_child) override;
+
+  // Retrieves the specified object's shortcut.
+  IFACEMETHODIMP get_accKeyboardShortcut(VARIANT var_id,
+                                         BSTR* access_key) override;
+
+  // Retrieves the name of the specified object.
+  IFACEMETHODIMP get_accName(VARIANT var_id, BSTR* name) override;
+
+  // Retrieves the IDispatch interface of the object's parent.
+  IFACEMETHODIMP get_accParent(IDispatch** disp_parent) override;
+
+  // Retrieves information describing the role of the specified object.
+  IFACEMETHODIMP get_accRole(VARIANT var_id, VARIANT* role) override;
+
+  // Retrieves the current state of the specified object.
+  IFACEMETHODIMP get_accState(VARIANT var_id, VARIANT* state) override;
+
+  // Gets the help string for the specified object.
+  IFACEMETHODIMP get_accHelp(VARIANT var_id, BSTR* help) override;
+
+  // Retrieve or set the string value associated with the specified object.
+  // Setting the value is not typically used by screen readers, but it's
+  // used frequently by automation software.
+  IFACEMETHODIMP get_accValue(VARIANT var_id, BSTR* value) override;
+  IFACEMETHODIMP put_accValue(VARIANT var_id, BSTR new_value) override;
+
+  // IAccessible methods not implemented.
+  IFACEMETHODIMP get_accSelection(VARIANT* selected) override;
+  IFACEMETHODIMP accSelect(LONG flags_sel, VARIANT var_id) override;
+  IFACEMETHODIMP get_accHelpTopic(BSTR* help_file,
+                                  VARIANT var_id,
+                                  LONG* topic_id) override;
+  IFACEMETHODIMP put_accName(VARIANT var_id, BSTR put_name) override;
+
+  AccessibilityRootNode();
+  ~AccessibilityRootNode();
+
+  void SetWindow(IAccessible* window);
+
+  void SetAlert(AccessibilityAlert* alert);
+
+  AccessibilityAlert* GetOrCreateAlert();
+
+ private:
+  // Helper method to redirect method calls to the contained window.
```

Comment:
This can also redirect to the alert, right? Consider:

```suggestion
  // Helper method to redirect method calls to the contained window or alert.
```